pkgs/development/libraries/qt-6/hooks/qtbase-setup-hook.sh

```diff
@@ -49,6 +49,20 @@ else # Only set up Qt once.
     }
     envBuildHostHooks+=(qmakePathHook)
 
+    export QTTOOLSPATH=
+
+    declare -Ag qttoolsPathSeen=()
+    qtToolsHook() {
+        # Skip this path if we have seen it before.
+        # MUST use 'if' because 'qttoolsPathSeen[$]' may be unset.
+        if [ -n "${qttoolsPathSeen[$1]-}" ]; then return; fi
+        qttoolsPathSeen[$1]=1
+        if [ -d "$1/libexec" ]; then
+            QTTOOLSPATH="${QTTOOLSPATH}${QTTOOLSPATH:+:}$1/libexec"
+        fi
+    }
+    envBuildHostHooks+=(qtToolsHook)
```

Comment:
Unsure about whether `BuildHost` is the right platform for this hook, we should double check it with strictDeps on. Also according to https://github.com/NixOS/nixpkgs/blob/6eeee6a7c8c8eaec03d06d9b8cef10599c727806/doc/stdenv/stdenv.chapter.md?plain=1#L1207, we actually should not be using `envBuildHostHooks` directly here (same for the qmake hook).

Apart from this, the implementation looks solid.

Reply:
`addEnvHooks "$hostOffset" qtToolsHook` seems work,  also check with `strictDeps = true` in qtdoc